tfexec/apply.go

```diff
@@ -127,21 +148,20 @@ func (tf *Terraform) ApplyJSON(ctx context.Context, w io.Writer, opts ...ApplyOp
 
 	tf.SetStdout(w)
 
-	cmd, err := tf.applyJSONCmd(ctx, opts...)
+	c := newApplyConfig(opts...)
+
+	cmd, err := tf.applyJSONCmd(ctx, c)
 	if err != nil {
 		return err
 	}
 
+	if c.gracefulShutdownConfig.Enable {
+		return tf.runTerraformCmdWithGracefulShutdown(ctx, cmd)
+	}
```

Comment:
Use the config value to decide whether to cancel pipe copy on context cancellation. Same for Destory.